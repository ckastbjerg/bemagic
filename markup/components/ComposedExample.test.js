/* eslint-env jest */
const path = require('path');
const ComposedExample = require('./ComposedExample');

test('returns a string of attributes from an object', () => {
    global.config = {
        namespace: 'prefix',
        componentsFolder: path.join(__dirname, '../../examples/src/components'),
    };

    expect(ComposedExample({ name: 'block' })).toMatchSnapshot();
});


test('will use derived example if no html file exists', () => {
    global.config = {
        namespace: 'prefix',
        componentsFolder: path.join(__dirname, '../../examples/src/components'),
    };

    expect(ComposedExample({ name: 'derive-example', derivedExample: '<div>A derived example</div>' })).toMatchSnapshot();
});
