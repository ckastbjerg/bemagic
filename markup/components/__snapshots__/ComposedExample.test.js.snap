exports[`test returns a string of attributes from an object 1`] = `
"
        <div class=\"bemagic-page__heading bemagic-page-heading\">
            Composed example
            <div class=\"bemagic-page-heading__explain\">
                extracted from /Users/ck/code/issuu/bemagic/examples/src/components/block/bemagic.html
            </div>
        </div>
        <div
            class=\"bemagic-page__section js-page-section \"
            data-class=\"section-prefix-block-full\"
        >
            <div class=\"ep-block\">
    Custom markup example...
    <div class=\"ep-block__element\">some element</div>
    <div class=\"ep-block__element--modifier\">some modifier</div>
</div>

        </div>
    "
`;

exports[`test will use derived example if no html file exists 1`] = `
"
        <div class=\"bemagic-page__heading bemagic-page-heading\">
            Derived example
            <div class=\"bemagic-page-heading__explain\">
                sophisticated guess regarding the components composition
            </div>
        </div>
        <div
            class=\"bemagic-page__section js-page-section \"
            data-class=\"section-prefix-derive-example-full\"
        >
            <div>A derived example</div>
        </div>
    "
`;
